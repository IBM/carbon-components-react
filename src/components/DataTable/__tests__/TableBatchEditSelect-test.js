import React from 'react';
import { shallow } from 'enzyme';
import { BatchEditableTableSelect } from '../';

describe('DataTable.BatchEditableTableSelect', () => {
  it('should render', () => {
    const wrapper = shallow(<BatchEditableTableSelect id="id-foo" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom attributes on the top-level node', () => {
    const wrapper = shallow(
      <BatchEditableTableSelect id="id-foo" data-foo="Foo" />
    );
    expect(wrapper.prop('data-foo')).toBe('Foo');
  });

  it('should wrap form validation message in a tooltip', () => {
    const wrapper = shallow(
      <BatchEditableTableSelect
        id="id-foo"
        invalid
        invalidText="invalid-text-foo"
      />
    );
    expect(wrapper.prop('invalidText').props.children).toBe('invalid-text-foo');
  });
});
