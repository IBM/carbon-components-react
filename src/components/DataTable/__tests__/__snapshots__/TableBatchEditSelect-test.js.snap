// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.BatchEditableTableSelect should render 1`] = `
<Select
  className="bx--select--batch-edit"
  disabled={false}
  helperText=""
  iconDescription="open list of options"
  id="id-foo"
  inline={false}
  invalid={false}
  invalidText={<InvalidIcon />}
  labelText=""
  light={false}
/>
`;
